Sources/OpenAPIRuntime/URICoder/Common/URIEncodedNode.swift

```diff
@@ -47,6 +47,16 @@ enum URIEncodedNode: Equatable {
         /// A date value.
         case date(Date)
     }
+
+    /// A primitive value or an array of primitive values.
+    enum PrimitiveOrArrayOfPrimitives: Equatable {
+
+        /// A primitive value.
+        case primitive(Primitive)
+
+        /// An array of primitive values.
+        case arrayOfPrimitives([Primitive])
```

Comment:
OOI, why do we need this additional nested enum. Can't we have this as a case in the outer enum?

Reply:
We could, but it'd force us to handle that case in several codepaths where we don't care about the distinction between "array" and "arrayOfPrimitives" (since most of this code does exhaustive matching of enum cases).

We also don't want to pass this enum value anywhere else, it's exclusively for the functions that only want to handle one of these two cases, verified at build time.